164829372 List Users

LarryKarani opened the pull request:
__What does this PR do?__
Enables authenticated users to get all the available authors and their profiles

__Description of the task to be completed__
Have the following endpoints working

__GET localhost/api/profiles/__
get the profiles of all authors

__GET localhost/api/users/__
get all authors

__How should this be manually tested?__

- Clone the repo and navigate to the root directory
- Create a virtual environment by typing` virtualenv venv` and activate it
- Install requirements by typing `pip install -r dev-requirements.txt`

Setup a database and define the DATABASE_URL in a .env file. Export the configuration variable
Run the tests by typing python `manage.py test`

__Relevant PT stories__
https://www.pivotaltracker.com/story/show/164829372

Commit:
[Feature 164829372] Get all profiles

Commit:
[Feature 16429372] Get all profiles

Commit:
[Finishes 164829372] Get all users

kihuha commented:
Don't you think it would be more appropriate to only have one level when retrieving all users? (What I mean by this is that instead of having to get the `user` dict then the `users` dict, we just have one dict level)
<img width="565" alt="Screenshot 2019-04-09 at 14 24 35" src="https://user-images.githubusercontent.com/29231722/55796752-645e1f00-5ad3-11e9-88fb-f226962c0850.png">

LarryKarani commented:
> Don't you think it would be more appropriate to only have one level when retrieving all users? (What I mean by this is that instead of having to get the `user` dict then the `users` dict, we just have one dict level)
> <img alt="Screenshot 2019-04-09 at 14 24 35" width="565" src="https://user-images.githubusercontent.com/29231722/55796752-645e1f00-5ad3-11e9-88fb-f226962c0850.png">

Thanks for the feedback. I'm fixing it right away

Commit:
Fix: Return list of all users

LarryKarani commented:
> Don't you think it would be more appropriate to only have one level when retrieving all users? (What I mean by this is that instead of having to get the `user` dict then the `users` dict, we just have one dict level)
> <img alt="Screenshot 2019-04-09 at 14 24 35" width="565" src="https://user-images.githubusercontent.com/29231722/55796752-645e1f00-5ad3-11e9-88fb-f226962c0850.png">

Its now fixed 

<img width="1052" alt="Screenshot 2019-04-09 at 14 35 17" src="https://user-images.githubusercontent.com/22986772/55797319-c5d2bd80-5ad4-11e9-9a57-203e9587dc2a.png">

WNjihia commented on authors/apps/profiles/tests/test_profiles.py:
```diff
@@ -93,6 +93,16 @@ def setUp(self):
         self.token = self.login.data['token']
         self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + self.token)
 
```
We could also test if an authenticated user can get user profiles

LarryKarani replied:
> We could also test if an authenticated user can get user profiles

This has already been tested in `test_profiles.py` from line 96 to 104

WNjihia reviewed (changes requested):
Hey some of the files in the list users pr are not related to the task. How come? It's out of scope for this particular pr
authors/apps/authentication/*
authors/apps/core/utils.py

WNjihia commented:
Also, there are merge conflicts. Kindly fix them

LarryKarani commented:
> Hey some of the files in the list users pr are not related to the task. How come? It's out of scope for this particular pr
> authors/apps/authentication/*
> authors/apps/core/utils.py

1. Authentication contains user views and tests. I added a method to get all the users in this view.

2. Inside the utils.py file, I added a method that overwrites the permissions classes so that I could specifically add permission to the get view and seclude the signup post method as they both were inside the same view.

LarryKarani commented:
> Also, there are merge conflicts. Kindly fix them

Okay

LarryKarani committed:
Merge branch 'develop' into ft-list-users-164829372

Commit:
Revert "Merge branch 'develop' into ft-list-users-164829372"

This reverts commit 0c67f1aed8fd79f8bb14aac1c74aa1721293775a, reversing
changes made to 54d39ba8a17a09ca7b39bc17d0455b4171d45679.

Commit:
Remove commented code

Commit:
Revert "Fix: Return list of all users"

This reverts commit 54d39ba8a17a09ca7b39bc17d0455b4171d45679.

Commit:
Revert "[Finishes 164829372] Get all users"

This reverts commit 91981da1a9812c10209ba5b7428a9c0fcb610690.

Commit:
Add get users view

Commit:
Revert "Add get users view"

This reverts commit 599f87d208f83fcefb0079e50095832ef45fdf67.
